src/components.rs

```diff
@@ -356,6 +358,69 @@ pub trait EventLike: Component {
         self.property_value("LOCATION")
     }
 
+    /// Set recurrence rules
+    #[cfg(feature = "recurrence")]
+    fn recurrence(&mut self, rruleset: RRuleSet) -> &mut Self {
+        let rrules = rruleset
+            .get_rrule()
+            .iter()
+            .map(ToString::to_string)
+            .collect::<Vec<_>>()
+            .join("\n");
+
+        let rdates = rruleset
+            .get_rdate()
+            .iter()
+            .map(|dt| dt.format("%Y%m%dT%H%M%SZ").to_string())
+            .collect::<Vec<_>>()
+            .join(",");
+
+        let exdates = rruleset
+            .get_exdate()
+            .iter()
+            .map(|dt| dt.format("%Y%m%dT%H%M%SZ").to_string())
+            .collect::<Vec<_>>()
+            .join(",");
+
+        self.add_property("RRULE", rrules)
+            .add_multi_property("RDATE", &rdates)
+            .add_multi_property("EXDATE", &exdates)
+    }
+
+    /// Get recurrence rules
+    #[cfg(feature = "recurrence")]
+    fn get_recurrence(&self) -> Option<RRuleSet> {
+        let dt_start_str = self.property_value("DTSTART")?;
+        let rrule_str = self.property_value("RRULE")?;
+
+        let mut rdates_str = self
+            .multi_properties()
+            .get("RDATE")
+            .unwrap_or(&vec![])
+            .iter()
+            .map(Property::value)
+            .collect::<Vec<_>>()
+            .join(",");
+        if !rdates_str.is_empty() {
+            rdates_str = format!("\nRDATE:{rdates_str}");
+        }
+
+        let mut exdates_str = self
+            .multi_properties()
+            .get("EXDATE")
+            .unwrap_or(&vec![])
+            .iter()
+            .map(Property::value)
+            .collect::<Vec<_>>()
+            .join(",");
+        if !exdates_str.is_empty() {
+            exdates_str = format!("\nEXDATE:{exdates_str}");
+        }
```

Comment:
Currently the implementation does not pass parameters of the properties to the `rrule`. Either the valid parameters within the `rrule` should be included or if possible all parameters should be passed to the `format!` macro for example by using the `Property::fmt_write`/`<Property as TryInto<String>::try_into` and removing the trailing `\r\n` or by adding a `impl Display for Property` which does not add a `\r\n`.

The problem is, that timestamps which include a time zone but as `TZID` parameter get rendered as a "floating" date-time and thus use the time zone of the local machine which is incorrect. E.g. an event with `DTSTART;TZID=Europe/Berlin:20251006T083000` will be rendered as `DTSTART:20251006T083000
RRULE:[...]` by this function.

Reply:
For example this works: https://github.com/ProbstDJakob/icalendar/commit/58d99eece50fc72070419e92744e359afd84841f

Reply:
For example this works: https://github.com/ProbstDJakob/icalendar/commit/58d99eece50fc72070419e92744e359afd84841f